packages/fetchai/skills/confirmation_aw1/handlers.py

```diff
@@ -118,6 +118,9 @@ def _handle_register(
             register_msg.info, register_msg.sender
         )
         if is_valid:
+            strategy.lock_registration_temporarily(
+                register_msg.sender, register_msg.info
+            )
```

Comment:
how the lock is restored when something wrong happen? E.g. I expected a try-except-finally pattern

Reply:
we now unlock on error. if an error happens in the ledger connection we don't handle it, it stays locked. ok for this simple agent.